### PR TITLE
DM-42628: A connection pool in the sync client API based on libcurl

### PR DESCRIPTION
**Pull request by iagaponenko**

### CODE REVIEW

File: src/http/ClientConfig.h [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    /// A value of a cert which would have to be pulled from the configuration
+    /// databases placed into a local file (at worker) be set with 'CURLOPT_PROXY_CAINFO'.
+    /// This option is used if it's impossible to preload required certificates
+    /// at workers, or make them directly readable by worker's ingest services otherwise.
+    static std::string const proxyCaInfoValKey;
+
+    /// The concurrency limit for the number of the asynchronous requests
+    /// to be processes simultaneously.
+    /// TODO: Move this parameter to the Replication System's Configuration
+    ///       as it doesn't belong here.
```
Comment (jgates108) [inline src/http/ClientConfig.h:87]: Maybe assign this TODO to a ticket.

File: src/http/ClientConnPool.h [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    /**
+     * Check for an error condition.
+     *
+     * @param scope A location from which the method was called (used for error reporting).
+     * @param errnum A result reported by the CURL library function.
+     * @throw std::runtime_error If the error-code is not CURLSHE_OK.
+     */
+    void _errorChecked(std::string const& scope, CURLSHcode errnum);
+
+    /// The miutex is shared by all instances of the pool.
```
Comment (jgates108) [inline src/http/ClientConnPool.h:74]: miutex -> mutex

**Review by jgates108 — APPROVED** [review]
Looks good.

File: src/http/ClientConfig.h (hunk 1 above)
Comment (iagaponenko) [reply src/http/ClientConfig.h:87]: Will do it.

**Pull request merged**